keras/layers/wrappers.py

```diff
@@ -287,25 +297,36 @@ def call(self, inputs, training=None, mask=None):
         if has_arg(self.layer.call, 'mask'):
             kwargs['mask'] = mask
 
-        y = self.forward_layer.call(inputs, **kwargs)
+        y = _to_list(self.forward_layer.call(inputs, **kwargs))
         y_rev = self.backward_layer.call(inputs, **kwargs)
         if self.return_sequences:
-            y_rev = K.reverse(y_rev, 1)
-        if self.merge_mode == 'concat':
-            output = K.concatenate([y, y_rev])
-        elif self.merge_mode == 'sum':
-            output = y + y_rev
-        elif self.merge_mode == 'ave':
-            output = (y + y_rev) / 2
-        elif self.merge_mode == 'mul':
-            output = y * y_rev
-        elif self.merge_mode is None:
-            output = [y, y_rev]
+            if self.return_state:
+                y_rev[0] = K.reverse(y_rev[0], 1)
+            else:
+                y_rev = K.reverse(y_rev, 1)
+        y_rev = _to_list(y_rev)
+
+        def merge_outputs(y, y_rev):
+            if self.merge_mode == 'concat':
+                return K.concatenate([y, y_rev])
+            elif self.merge_mode == 'sum':
+                return y + y_rev
+            elif self.merge_mode == 'ave':
+                return (y + y_rev) / 2
+            elif self.merge_mode == 'mul':
+                return y * y_rev
+
+        inner_layer_tensors = y + y_rev
+        if self.merge_mode is None:
+            output = inner_layer_tensors
+        else:
+            output = [merge_outputs(x1, x2) for x1, x2 in zip(y, y_rev)]
```

Comment:
I don't think it makes sense to concatenate/multiply/etc RNN states. The state of a bidirectional LSTM is simply the list of the states of the underlying RNNs. This list, passed to `call`, should then be split and routed to each appropriate RNN.